src/components/Table/Table.tsx

```diff
@@ -119,44 +125,207 @@ const Tbody = styled.tbody`
     gap: 0.25rem;
   }
 `;
+
+const SelectData = styled.td`
+  overflow: hidden;
+  ${({ theme }) => `
+    color: ${theme.click.table.row.color.text.default};
+    font: ${theme.click.table.cell.text.default};
+    padding: ${theme.click.table.body.cell.space.md.y} ${theme.click.table.body.cell.space.md.x};
+  `}
+  @media (max-width: 768px) {
+    width: auto;
+    align-self: stretch;
+    position: absolute;
+    left: 0;
+    top: 0;
+    bottom: 0;
+    ${({ theme }) => `
+      padding: ${theme.click.table.body.cell.space.sm.y} ${theme.click.table.body.cell.space.sm.x};
+      border-right: 1px solid ${theme.click.table.row.color.stroke.default};
+    `}
+  }
+`;
+
+const TableWrapper = styled.div`
+  width: 100%;
+  height: 100%;
+  overflow: auto;
+`;
+
+const TableOuterContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: 1rem;
+`;
+
+const MobileActions = styled.div`
+  display: none;
+  @media (max-width: 768px) {
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+    padding: 0 ${({ theme }) => theme.click.table.body.cell.space.sm.x};
+  }
+`;
+
 interface TableCellType extends HTMLAttributes<HTMLTableCellElement> {
   label: ReactNode;
 }
-type TableRowType = Array<TableCellType>;
-interface TableProps extends HTMLAttributes<HTMLTableElement> {
+interface TableRowType
+  extends Omit<HTMLAttributes<HTMLTableRowElement>, "onSelect" | "id"> {
+  id: string | number;
+  items: Array<TableCellType>;
+}
+
+interface CommonTableProps
+  extends Omit<HTMLAttributes<HTMLTableElement>, "children" | "onSelect"> {
   headers: Array<TableHeaderProps>;
   rows: Array<TableRowType>;
 }
-const Table = forwardRef<HTMLTableElement, TableProps>(
-  ({ headers, rows, ...props }, ref) => (
-    <StyledTable
-      ref={ref}
-      {...props}
+
+type SelectReturnValue = {
+  id: string | number;
```

Comment:
isn't the id already included in the item?

Reply:
It is @fneves suggested to add it separately in the comment above
So added it

Reply:
Fixed it by passing index
I misunderstood the initial comment